File: java/src/test/java/it/xpug/kata/birthday_greetings/AcceptanceTest.java
package it.xpug.kata.birthday_greetings;

import static org.junit.Assert.*;

import org.junit.*;

import com.dumbster.smtp.*;

import java.util.List;

public class AcceptanceTest {

	private static final int NONSTANDARD_PORT = 9999;
	private BirthdayService birthdayService;
	private SimpleSmtpServer mailServer;

	@Before
	public void setUp() throws Exception {
		mailServer = SimpleSmtpServer.start(NONSTANDARD_PORT);
		birthdayService = new BirthdayService();
	}

	@After
	public void tearDown() throws Exception {
		mailServer.stop();
		Thread.sleep(200);
	}

	@Test
	public void willSendGreetings_whenItsSomebodysBirthday() throws Exception {

		birthdayService.sendGreetings("employee_data.txt", new XDate("2008/10/08"), "localhost", NONSTANDARD_PORT);

		assertEquals("message not sent?", 1, mailServer.getReceivedEmails().size());
		SmtpMessage message = (SmtpMessage) mailServer.getReceivedEmails().get(0);
		assertEquals("Happy Birthday, dear John!", message.getBody());
		assertEquals("Happy Birthday!", message.getHeaderValue("Subject"));
		List<String> recipients = message.getHeaderValues("To");
		assertEquals(1, recipients.size());
		assertEquals("john.doe@foobar.com", recipients.get(0));
	}

	@Test
	public void willNotSendEmailsWhenNobodysBirthday() throws Exception {
		birthdayService.sendGreetings("employee_data.txt", new XDate("2008/01/01"), "localhost", NONSTANDARD_PORT);

		assertEquals("what? messages?", 0, mailServer.getReceivedEmails().size());
	}
}

File: csharp/BirthdayGreetingsTests/AcceptanceTest.cs
using NUnit.Framework;
using System;
using BirthdayGreetings;
using netDumbster.smtp;

namespace BirthdayGreetingsTests
{
	[TestFixture]
	public class AcceptanceTest
	{
		private BirthdayService service;
		private string FILE_NAME = "../../../BirthdayGreetings/employee_data.txt";
		private SimpleSmtpServer smtpServer;

		[SetUp]
		public void setUp() 
		{
			service = new BirthdayService ();
			smtpServer = SimpleSmtpServer.Start ();
		}

		[TearDown]
		public void TearDown()
		{
			smtpServer.Stop();
		}

		[Test]
		public void willSendGreetings_whenItsSomebodysBirthday() 
		{
			service.SendGreetings(FILE_NAME, new XDate("2008/10/08"), "localhost", smtpServer.Port);

			Assert.AreEqual(1, smtpServer.ReceivedEmailCount, "message not sent?");
			var message = smtpServer.ReceivedEmail [0];
			Assert.AreEqual ("Happy Birthday, dear John!", message.MessageParts[0].BodyData);
			Assert.AreEqual ("Happy Birthday!", message.Headers.Get("subject"));
			Assert.AreEqual (1, message.ToAddresses.Length);
			Assert.AreEqual ("john.doe@foobar.com", message.ToAddresses[0].ToString());
		}

		[Test]
		public void willNotSendEmailsWhenNobodysBirthday() 
		{
			service.SendGreetings(FILE_NAME, new XDate("2008/01/01"), "localhost", smtpServer.Port);
			Assert.AreEqual(0, smtpServer.ReceivedEmailCount, "what? messages?");
		}

	}
}

